### PR TITLE
feat!: switch HDRP to URP

### PR DESCRIPTION
**Pull request by Aysenayilmaz**
Render pipeline has been switched from HDRP to URP. Adjustments have been made in the TrafficLight script and project settings to be compatible with URP materials.

Related issue :

- https://github.com/tier4/AWSIM/issues/261#issuecomment-1979578888
- https://github.com/autowarefoundation/AWSIM/issues/10

### CODE REVIEW

**Comment by xmfcx** [comment]
@Aysenayilmaz thanks for the PR!
Please update the documentation links with the new nishishinjuku environment link.

And share the steps to review this task.

When this is ready to review, please click this button.
Screenshot from 2024-03-07 15-21-00

**Comment by mozhoku** [comment]
Hello @Aysenayilmaz . If we are switching to URP, there is no need for HDRP packages. I suggest that it should be removed from the project altogether since they can't work together as stated [here](https://docs.unity3d.com/Packages/com.unity.render-pipelines.universal@7.1/manual/faq.html#:~:text=Can%20I%20use%20URP%20and,and%20light%20models%20are%20different.).

Here is the link for the how to do it from the Unity Package Manager:
- https://docs.unity3d.com/Manual/upm-ui-remove.html

**Comment by xmfcx** [comment]
I was wondering if it's possible to keep them both for separate scenes.

Like, for synthetic data generation, keeping them for a scene dedicated for this.

And for normal day to day simulation, use URP scenes.

**Comment by mozhoku** [comment]
> I was wondering if it's possible to keep them both for separate scenes.
> 
> Like, for synthetic data generation, keeping them for a scene dedicated for this.
> 
> And for normal day to day simulation, use URP scenes.

You can give it a try. I don't think it is a good idea to have two of them in the same project. You would need to change active pipeline from the settings every time you switch between scenes afaik. If you want to keep it, there is no need to rush for the removal. We can remove it later on.

File: docs/GettingStarted/SetupUnityProject/index.md
Comment (mozhoku) [inline docs/GettingStarted/SetupUnityProject/index.md:1]: If we are keeping the HDRP, these steps should be separate from the HDRP version.

File: Assets/AWSIM/Scripts/Environments/TrafficLight.cs [hunk 1]
```diff
@@ -102,15 +102,14 @@ public class EmissionConfig
...
-            const string LightOnFlag = "_LightOn";
+            const string EmissionColor = "_EmissionColor";
+            const string EmissionIntensity = "_EmissionIntensity";
+            const string EmissionExposureWeight = "_EmissionExposureWeight";
             const float flashIntervalSec = 0.5f;                // flash bulb lighting interval(sec).
 
             float timer = 0;                            // used for flashing status.     NOTE: Might as well make it static and refer to the same time. 
-            Color defaultEmissiveColor;                 // default bulb material emissive color.
-            float defaultEmissiveExposureWeight;        // default bulb mateiral emissive exposure weight
+            Color defaultEmissionColor;                 // default bulb material  color.
```
Comment (mozhoku) [inline Assets/AWSIM/Scripts/Environments/TrafficLight.cs:111]:
```suggestion
            Color defaultEmissionColor;                 // default bulb material color.
```

File: Assets/AWSIM/Scripts/Environments/TrafficLight.cs [hunk 2]
```diff
@@ -102,15 +102,14 @@ public class EmissionConfig
...
+            float defaultEmissionExposureWeight;        // default bulb mateiral  exposure weight
```
Comment (mozhoku) [inline Assets/AWSIM/Scripts/Environments/TrafficLight.cs:112]:
```suggestion
            float defaultEmissionExposureWeight;        // default bulb material exposure weight
```

**Review by mozhoku — COMMENTED** [review]
Since it's decided that we are keeping both pipelines. There should be two separate AutowareSimulation.unity scenes for URP and HDRP respectively. 🤗

**Comment by xmfcx** [comment]
After reading:
- https://www.reddit.com/r/Unity3D/comments/11u4cvf/it_is_a_good_idea_to_install_urp_and_hdrp_in_the/
- https://www.reddit.com/r/Unity3D/comments/rry7al/can_i_use_hdrp_and_urp_in_the_same_project/
- https://www.reddit.com/r/Unity3D/comments/n6w8z5/can_one_project_support_both_urp_and_hdrp/
- https://www.reddit.com/r/Unity3D/comments/m1u7zd/developing_the_same_code_in_urp_android_and_hdrp/

I've concluded that supporting both in the same project is not feasible.

Let's remove all references to the HDRP from this project.

For synthetic data generation (for cameras), I'd suggest creating a separate project, I don't know.

This is not a high priority for now from AWF perspective.

**Comment by Aysenayilmaz** [comment]
### testing steps:

1. Cloning this repo from https://github.com/Aysenayilmaz/AWSIM/tree/feat/switch-to-urp
2. Document should be followed https://github.com/Aysenayilmaz/AWSIM/blob/feat/switch-to-urp/docs/GettingStarted/SetupUnityProject/index.md

**Comment by mozhoku** [comment]
@Aysenayilmaz I've switched to your branch and opened the project. Sensor kit is pink. Please update it to URP. Other than that seems okay. 👍 

AWSIM_YlLbbMPBAM

**Comment by mozhoku** [comment]
It looks okay now. 👍 
image

**Comment by xmfcx** [comment]
@mozhoku could you export an executable and test it with Autoware?

**Comment by mozhoku** [comment]
I've tested the executable with Autoware. There wasn't any issues with driving.

Here is the link:
https://youtu.be/upBNMPwA-UM

**Review by xmfcx — APPROVED** [review]

**Pull request merged**